test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,341 @@
+						{
+							Name:  name,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createServeHostnameReplicationController(c *client.Client, ns, name string, replicas int) (*api.ReplicationController, error) {
+	By(fmt.Sprintf("creating replication controller %s", name))
+	return c.ReplicationControllers(ns).Create(newReplicationControllerWithNameSelector(name, replicas, "gcr.io/google_containers/serve_hostname:1.1"))
+}
+
+func resizeReplicationController(c *client.Client, ns, name string, replicas int) error {
+	rc, err := c.ReplicationControllers(ns).Get(name)
+	if err != nil {
+		return err
+	}
+	rc.Spec.Replicas = replicas
+	_, err = c.ReplicationControllers(rc.Namespace).Update(rc)
+	return err
+}
+
+func waitForPodsCreated(c *client.Client, ns, name string, replicas int) (*api.PodList, error) {
+
+	// List the pods, making sure we observe all the replicas.
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	var pods *api.PodList
+	for start := time.Now(); time.Since(start) < time.Minute; time.Sleep(5 * time.Second) {
+		var err error
+		pods, err = c.Pods(ns).List(label, fields.Everything())
+		if err != nil {
+			return nil, err
+		}
+
+		Logf("Controller %s: Found %d pods out of %d", name, len(pods.Items), replicas)
+		if len(pods.Items) == replicas {
+			return pods, nil
+		}
+	}
+	return nil, fmt.Errorf("controller %s: Gave up waiting for %d pods to come up", name, replicas)
+}
+
+func waitForPodsRunning(c *client.Client, pods *api.PodList) []error {
+	// Wait for the pods to enter the running state. Waiting loops until the pods
+	// are running so non-running pods cause a timeout for this test.
+	By("ensuring each pod is running")
+	e := []error{}
+	for _, pod := range pods.Items {
+		// TODO: make waiting parallel.
+		err := waitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
+		if err != nil {
+			e = append(e, err)
+		}
+	}
+	return e
+}
+
+func verifyPodsResponding(c *client.Client, ns, name string, pods *api.PodList) error {
+	By("trying to dial each unique pod")
+	retryTimeout := 2 * time.Minute
+	retryInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	return wait.Poll(retryInterval, retryTimeout, podResponseChecker{c, ns, label, name, pods}.checkAllResponses)
+}
+
+var _ = Describe("ResizeNodes", func() {
+	supportedProviders := []string{"gce"}
+	var testName string
+	var c *client.Client
+	var ns string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		testingNs, err := createTestingNS("resize-nodes", c)
+		ns = testingNs.Name
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if !providerIs(supportedProviders...) {
+			return
+		}
+		By(fmt.Sprintf("destroying namespace for this suite %s", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete namespace %v", err)
```

Comment:
nit: I think you want a ":" after "namespace". You could also include the namespace in there, like

```go
Failf("Couldn't delete namespace '%s': %v", ns, err)
```